### PR TITLE
Prevent assets loading in chunks

### PR DESCRIPTION
**Pull request by jxom**
Fixes RNBW-2143

## What changed (plus any additional context for devs)

This PR fixes the jarring "loading tokens in chunks" UI when initializing a wallet for the first time.

Since we are dealing with websockets to retrieve asset data, this makes loading state a bit cumersome to manage. In the ideal world, it would be nice if loading assets would be sync, but since websockets are naturally async, this implementation will have to do for now (unless we want to show a "refetching" floating pill instead of skeleton). 

The implementation involves setting a "debounced" function (with a specified timeout) that sets the loading state to false when we no longer receive any more "received asset" events from the websocket. This only occurs for a wallet loading their assets for the first time (where we see the jank). However, on subsequent wallet loads, it just updates in the background.

## PoW (screenshots / screen recordings)

Before:

https://www.loom.com/share/029baaaf84c74de297cb75e5871900dc

After:

https://www.loom.com/share/49efb12e0ff24cd1b065d502564564b7

## Dev checklist for QA: what to test

1. Open a fresh instance of Rainbow (with no wallets imported)
2. Import accounts with a large number of assets (e.g. "rainbowwallet.eth")

### CODE REVIEW

File: src/redux/data.ts [hunk 1]
```diff
@@ -1820,3 +1857,49 @@ export default (state: DataState = INITIAL_STATE, action: DataAction) => {
       return state;
   }
 };
+
+// -- Middlewares ---------------------------------------- //
+
+const FETCHING_TIMEOUT = 10000;
```
[the remark this answers is not in the document]
Comment (jxom) [reply src/redux/data.ts:1863]: posting slack thread here for context:
https://rainbowhaus.slack.com/archives/C02C8JQ313N/p1649718532564969

[the remark this answers is not in the document]
Comment (jxom) [reply src/redux/data.ts:1863]: yo @brunobar79 @welps – are we happy w/ this implementation for now?

[the remark this answers is not in the document]
Comment (welps) [reply src/redux/data.ts:1863]: Sorry, was I blocking this?

**Review by welps — APPROVED** [review]

File: src/redux/data.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jxom) [reply src/redux/data.ts:1863]: not really, just need some approvals!

**Review by skylarbarrera — APPROVED** [review]
this is very nice ty 🙏

**Review by ibrahimtaveras00 — CHANGES_REQUESTED** [review]
I'm seeing behavior that when I switch in between vitalik.eth (watching) and rainbowwallet.eth (owner) a lot of vitalik.eth assets are showing up instead; seen here = https://recordit.co/tdUiy4Avay

Screen Shot 2022-04-29 at 12 43 07 PM

steps:
- fresh install (0 wallets) 
- watch vitalik.eth 
- import rainbowwallet.eth 
- switch in between the 2 accounts

**Comment by jinchung** [comment]
status update: reached out to jake, ibrahim, skylar for next steps and verifying token wallet mixup still an issue

**Comment by jxom** [comment]
@ibrahimtaveras00 – try now?

**Review by ibrahimtaveras00 — APPROVED** [review]
Looks good now, QA Passed 👍🏽

**Pull request merged**